zip-plus/src/code/iprs/pntt/radix8/butterfly.rs

```diff
@@ -0,0 +1,72 @@
+use num_traits::CheckedAdd;
+use zinc_utils::add;
+
+use super::MulByTwiddle;
+
+/// Generate the indices of twiddle factors
+/// used for a butterfly.
+///
+/// In case of radix 8, we have 8 butterflies.
+macro_rules! generate_radix_8_butterfly {
+    ($i:literal) => {
+        [
+            0,
+            $i % 8,
+            (2 * $i) % 8,
+            (3 * $i) % 8,
+            (4 * $i) % 8,
+            (5 * $i) % 8,
+            (6 * $i) % 8,
+            (7 * $i) % 8,
+        ]
+    };
+    ($i:expr) => {
+        match $i {
+            0 => generate_radix_8_butterfly!(0),
+            1 => generate_radix_8_butterfly!(1),
+            2 => generate_radix_8_butterfly!(2),
+            3 => generate_radix_8_butterfly!(3),
+            4 => generate_radix_8_butterfly!(4),
+            5 => generate_radix_8_butterfly!(5),
+            6 => generate_radix_8_butterfly!(6),
+            7 => generate_radix_8_butterfly!(7),
+            _ => panic!("Incorrect butterfly number"),
+        }
+    };
+}
```

Comment:
If this is just 8 * 8 table, how about we generate it at compile time rather than at runtime?
```suggestion
const BUTTERFLY: [[usize; 8]; 8] = {
    macro_rules! generate_radix_8_butterfly {
        ($i:expr) => {
            [
                0,
                $i % 8,
                (2 * $i) % 8,
                (3 * $i) % 8,
                (4 * $i) % 8,
                (5 * $i) % 8,
                (6 * $i) % 8,
                (7 * $i) % 8,
            ]
        };
    }
    let mut result = [[0; 8]; 8];
    const_for!((i in 0..(result.len())) {
        result[i] = generate_radix_8_butterfly!(i);
    });
    result
};
```

Reply:
I got rid of it completely in https://github.com/NethermindEth/zinc-plus/pull/53. So I suggest we don't touch it in this PR.

Reply:
Actually let's make the git history slightly nicer